### PR TITLE
Format and CI

### PR DESCRIPTION
**Pull request by Dale-Koenig**
Set up clang format to run in ci and format all files.  If this is merged and you have in progress branches, it is recommended to take the `.clang-format` file from here, run `format.sh` to format all files, commit, and THEN merge rr_devel into your branch

### CODE REVIEW

File: check_clang_format.sh
Comment (david-simoes-93) [inline check_clang_format.sh:39]: :+1:

File: format_cpp.sh
Comment (david-simoes-93) [inline format_cpp.sh]: in the future, we might change this to also add [a newline at the end of all files](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline), but let's leave it for another PR

**Review by david-simoes-93 — APPROVED** [review]
LGTM

**Comment by Dale-Koenig** [comment]
I will switch to clang format 10 before merging

@veerajsk waiting on approval

**Review by veerajsk — APPROVED** [review]

**Pull request merged**
